src/Illuminate/Foundation/Console/RouteCacheCommand.php

```diff
@@ -30,13 +45,15 @@ class RouteCacheCommand extends Command {
 	/**
 	 * Create a new route command instance.
 	 *
-	 * @param  \Illuminate\Filesystem\Filesystem  $files
-	 * @return void
+	 * @param  \Illuminate\Filesystem\Filesystem $files
+	 * @param  \Illuminate\Routing\Router  $router
 	 */
-	public function __construct(Filesystem $files)
+	public function __construct(Filesystem $files, Router $router)
 	{
 		parent::__construct();
 
+		$this->router = $router;
+		$this->routes = $router->getRoutes();
```

Comment:
Unfortunately that isn't working right. After i wake up I'll push an
example app to github that demonstrates the problem. Hopefully we can work
together to find a solution.

On Friday, February 20, 2015, Mior Muhammad Zaki <notifications@github.com>
wrote:

> In src/Illuminate/Foundation/Console/RouteCacheCommand.php
> <https://github.com/laravel/framework/pull/7523#discussion_r25049793>:
>
> >  	{
> >  		parent::__construct();
> >
> > +		$this->router = $router;
> > +		$this->routes = $router->getRoutes();
>
> If you are loading from current request, which is before running
> route:clear, you simply loading available routes from cached, this is why
> @taylorotwell <https://github.com/taylorotwell> load routes from a fresh
> application.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laravel/framework/pull/7523/files#r25049793>.
>